wiki/SSoT_deutsch

```diff
@@ -0,0 +1,637 @@
+4. Verwenden wir das XML als SSOT?
+_Nein, wir rendern aus der SSOT die Modelle._
+_Für z.B. BPMN rendern wir SSOT zu XML zu BPMN?_
+5. Wie speichern wir die Kanten aus BPMN?
+_Implizit durch die Nachfolger/Vorgänger-Beziehungen. Die genauen Waypoints müssen "errechnet" werden._
+6. Was für Konsequenzen hat das nicht-Speichern von Modellierungsattributen für unser Projekt? (**BPMN-spezifisch**)
+    - exakte Anordnung im Interface wird nicht beibehalten;
+jedoch: Unterstützung beim Einhalten des Standards (automatischer Pretty Print)
+    - Es wir eine Logik für das Updaten der SSoT nach Löschen (und Hinzufügen) benötigt
+    - Config-Dateien mit Größen, Farben, Symbolen etc. werden für jedes Modellierungsinterface benötigt
+
+## Anwendungskonzept SSoT
+
+**Feststellung: SSoT sollte näher an den grundlegenden Programmierkonstrukten sein als an einem BPMN-Modell.**
+Daher: Alle Interfaces werden in ein Graphen-Modell gerendert, bei denen die Knoten durch folgende Tags bestimmt werden: `SEQUENCE`, `BRANCH`, `LOOP` und `TRIGGER`. Diese Tags bilden die grundsätzlichen Programmierkonstrukte ab, die auch im **Robot Framework** verwendet werden können.
+
+
+### Übersicht
+|  SSoT-Tag   |      BPMN-Repräsentation       |  Code-Repräsentation   |      ArkA 0.3      |      ArkA 1.0      |
+|:-----------:|:------------------------------:|:----------------------:|:------------------:|:------------------:|
+| INSTRUCTION |           Aktivität            |   "eine Zeile Code"    | :heavy_check_mark: | :heavy_check_mark: |
+|    CASE     |          XOR-Gateway           | Verzweigung (IF, CASE) |        :x:         | :heavy_check_mark: |
+|    LOOP     | XOR-Split & Join (Kombination) | Schleife (FOR, WHILE)  |        :x:         | :heavy_check_mark: |
+|   MARKER    |       Start- & End-Event       |           -            | :heavy_check_mark: | :heavy_check_mark: |
+| _EXCEPTION_ |      _intermediate Event_      |      _TRY-CATCH_       |        :x:         |        :x:         |
+
+**Besonderheiten bei Events:**
+**Timer:** Langfristig sollen Timer in BPMN als Zwischenevent dargestellt werden. Momentan ist beispielhaft ein "Sleep-Event" (zum Pausieren der Ausführung) als Instruction mit einer RPA-Aktion zu definieren.
+
+**Event-Typen:** Diese werden nicht gespeichert. Events erscheinen daher immer als "leere Kreise". Sie sind auch nicht mit Logik verknüpfbar. Soll z.B. ein End-Event eine Mail senden, so ist dieses Konstrukt bitte als `Aktivität: Mail Senden` und `End Event` zu modellieren.
+Es werden momentan keine Zwischenevents unterstützt.
+
+### Übersicht SSOT Aufbau
+![](https://i.imgur.com/NZkjzCg.png)
+
+
+### Anwendung: HEADER
+
+Hierbei ändert sich nichts im Vergleich zu Daniels Aufschlag einer SSoT. Nutzer-Informationen zu einem Robot (Ersteller, Freigegeben für etc.) werden in einer zusätzlichen Datenbank abgelegt.
```

Comment:
Sollte so formuliert werden: "... werden in einer zusätzlichen **Datei in der** Datenbank abgelegt"